rapid7_insightvm/komand_rapid7_insightvm/util/resource_helpers.py

```diff
@@ -0,0 +1,182 @@
+            self.session.headers.pop('nexposeCCSessionID')
+            self.session.headers.pop('Cookie')
+        else:
+            error = "ERROR: Failed to log out via APIv1"
+            raise Exception(error)
+
+
+class ValidateUser:
+    """
+    Validates that a user exists and has the correct permissions
+    """
+    def __init__(self, session, logger):
+        """
+        Creates a new instance of ValidateUser
+        :param session: Session object available to Komand actions/triggers, usually self.connection.session
+        :param logger: Logger object available to Komand actions/triggers, usually self.logger
+        :return: ResourceHelper object
+        """
+        self.logger = logger
+        self.session = session
+        self.requests = resource_requests.ResourceRequests(session=self.session, logger=self.logger)
+
+        # Suppress insecure request messages
+        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+    def validate_role_exists(self, console_url: str, role_id: str) -> dict:
+        """
+        Validate that a provided InsightVM role exists
+        :param console_url: URL to the InsightVM console
+        :param role_id: ID of the role to validate
+        :return: The role provided that it exists, if it does not exist a detailed error is raised
+        """
+        endpoint = endpoints.Role.roles(console_url, role_id)
+        try:
+            response = self.requests.resource_request(endpoint=endpoint)
+            return response
+        except ResourceNotFound:
+            # Get roles to provide a better error
+            endpoint = endpoints.Role.roles(console_url)
+            roles = self.requests.resource_request(endpoint=endpoint)['resources']
+            role_ids = [r['id'] for r in roles]
+
+            error = f"ERROR: Specified role ID ({role_id}) does not exist, valid role IDs: {role_ids}"
+            self.logger.error(error)
+            raise ResourceNotFound(error)
+
+    def validate_user_permissions(self, console_url: str, user: dict) -> dict:
+        """
+        Validate that user permissions are appropriate given the user's role.
+        This is useful when updating a user's role or create a new user as
+        some roles required the allSites, allAssetGroups, or superuser parameters
+        to be set.
+        :param console_url: URL to the InsightVM console
+        :param user: dict containing the details for a user (from the InsightVM API or matching the format)
+        :return: The user dict with updated permissions if required
+        """
+        endpoint = endpoints.Role.roles(console_url, user['role']['id'])
+        role = self.requests.resource_request(endpoint=endpoint)
+
+        # Set All Sites permission if required and not set
+        if not user['role']['allSites']:
+            for permission in ("all-permissions", "manage-sites", "manage-tags"):
```

Comment:
```suggestion
            for permission in ["all-permissions", "manage-sites", "manage-tags"]:
```
Did you really mean to use a tuple here?

Reply:
I don't know this is not my code. I just moved it here because It dose not belong in the api requests class